### ISSUE TITLE
DispersionAmmo get stuck in water

### ISSUE THREAD

**Issue by Reorn** [OldUnreal/Unreal-testing#358]
In a large water pool DispersionAmmo lose speed, slow down, and then stop in the water.
However, they do not disappear.
Last Beta (2024-03 and older betas).
Test map attached.
![TestMap](https://github.com/OldUnreal/Unreal-testing/assets/108881415/904877c0-c092-47ca-b6b0-d9e9892b98e8)
[TestMap.zip](https://github.com/OldUnreal/Unreal-testing/files/14653660/TestMap.zip)

**Comment by TheQuickFox** [comment]
Very old bug. Devs are aware of this at some point I read that they decided not to fix this because it would cause issues with older content/mods that derive of the dispersion pistol projectile.

All projectiles have a max lifetime setting after which the projectile self-destructs. But for the dispersion pistol projectile this lifetime is reset in a loop causing it to never reach the timeout.

This issue can be slightly misused on servers by shooting a gazillion of DP projectiles underwater. The server has to keep track of them and the client has to render them. The rendering may cause FPS slowdowns on other clients, especially if they have older computers. But it takes very long to reach this state and usually the server will switch maps before this is reached.

Issue may be worse when the server has custom weapons that spawn gazillions of DP projectiles.

**Comment by Reorn** [comment]
The problem I see is that the particle loses speed in the water even though it definition is supposed to be moving at the CONSTANT speed.
This is a global problem for all particles.

LifeSpan is updated in tick.

simulated function Tick(float DeltaTime)
{
	if ( Physics != PHYS_None )
		LifeSpan = 0.7; // keep resetting it - can't make it longer since animspriteeffect base of this
}

I see no reason why this can't be done:

simulated function Tick(float DeltaTime)
{
 if ( Physics != PHYS_None ) LifeSpan = 0.7;
 Velocity = Speed * vector(Rotation);
}

However, it is better to solve this problem at the native code for ALL particles.

**Comment by TheQuickFox** [comment]
Yeah, I think Epic once implemented this to add some realism to the game by slowing down projectiles due to water resistance. It would be fine with me if this effect is removed from the game as long as it does not cause mods to stop working.
